src/main/java/com/marginallyclever/makelangelo/plotter/settings/PlotterSettings.java

```diff
@@ -157,41 +169,53 @@ public boolean isRegistered() {
-
-		startingPositionIndex = Integer.parseInt(uniqueMachinePreferencesNode.get("startingPosIndex", Integer.toString(startingPositionIndex)));
-
-		int r, g, b;
-		r = uniqueMachinePreferencesNode.getInt("paperColorR", paperColor.getRed());
-		g = uniqueMachinePreferencesNode.getInt("paperColorG", paperColor.getGreen());
-		b = uniqueMachinePreferencesNode.getInt("paperColorB", paperColor.getBlue());
+		int r = thisMachineNode.getInt("paperColorR", paperColor.getRed());
+		int g = thisMachineNode.getInt("paperColorG", paperColor.getGreen());
+		int b = thisMachineNode.getInt("paperColorB", paperColor.getBlue());
 		paperColor = new ColorRGB(r, g, b);
 
 		// setCurrentToolNumber(Integer.valueOf(uniqueMachinePreferencesNode.get("current_tool",
 		// Integer.toString(getCurrentToolNumber()))));
-		isRegistered = Boolean.parseBoolean(uniqueMachinePreferencesNode.get("isRegistered", Boolean.toString(isRegistered)));
-
-		loadPenConfig(uniqueMachinePreferencesNode);
-		hardwareName = uniqueMachinePreferencesNode.get("hardwareVersion", hardwareName);
+		isRegistered = Boolean.parseBoolean(thisMachineNode.get("isRegistered", Boolean.toString(isRegistered)));
+		hardwareName = thisMachineNode.get("hardwareVersion", hardwareName);
+
+		blockBufferSize 		= thisMachineNode.getInt("blockBufferSize", blockBufferSize);
+		segmentsPerSecond 		= thisMachineNode.getInt("segmentsPerSecond", segmentsPerSecond);
+		minSegmentLength 		= thisMachineNode.getDouble("minSegmentLength", minSegmentLength);
+		minSegTime 				= thisMachineNode.getLong("minSegTime", minSegTime);
+		handleSmallSegments 	= thisMachineNode.getBoolean("handleSmallSegments", handleSmallSegments);
+		minAcceleration			= thisMachineNode.getDouble("minAcceleration", minAcceleration);
+		minimumPlannerSpeed 	= thisMachineNode.getDouble("minimumPlannerSpeed", minimumPlannerSpeed);
+		
+		loadJerkConfig(thisMachineNode);
+		loadPenConfig(thisMachineNode);
+	}
+	
+	private void loadJerkConfig(Preferences thisMachineNode) {
+		Preferences jerkNode = thisMachineNode.node("maxJerk");
+		for(int i=0;i<maxJerk.length;i++) {
+			maxJerk[i] = jerkNode.getDouble(Integer.toString(i), maxJerk[i] );
+		}
 	}
 
-	protected void loadPenConfig(Preferences prefs) {
+	private void loadPenConfig(Preferences prefs) {
 		prefs = prefs.node("Pen");
-		setPenDiameter(Double.parseDouble(prefs.get("diameter", Double.toString(penDiameter))));
-		setPenLiftTime(Double.parseDouble(prefs.get("z_rate", Double.toString(penLiftTime))));
-		setPenDownAngle(Double.parseDouble(prefs.get("z_on", Double.toString(penDownAngle))));
-		setPenUpAngle(Double.parseDouble(prefs.get("z_off", Double.toString(penUpAngle))));
-		setTravelFeedRate(Double.parseDouble(prefs.get("feed_rate", Double.toString(travelFeedRate))));
-		setDrawFeedRate(Double.parseDouble(prefs.get("feed_rate_current", Double.toString(drawFeedRate))));
+		setPenDiameter(		prefs.getDouble("diameter"			, penDiameter	));
```

Comment:
The formatting does not looks like a table :'(

Reply:
Fucking tabs...